### PR TITLE
Branch p-find

### LINKED ISSUE #6: Find students under a module
As a TA for different modules, I want to be able to organise student information according to module so that I can manage my tasks in an organised manner

### LINKED ISSUE #15: Add feature to search students by module
Update the functionality for 'p-find' so that user can search students by module

### PR DESCRIPTION
**Pull request by yeo-yiheng**
Fixes #15  
Fixes #6 

Add functionality to find students by module and name at the same time, or one at a time. Also add the validity checker for proper module code, and certain corner cases for misuse of command. Finally, add tests to fully test the functions of the find command to ensure that it is fully working.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -1,33 +1,154 @@
...
+
+        Pattern namePattern = Pattern.compile(namePrefix.getPrefix());
+        Pattern modulePattern = Pattern.compile(modulePrefix.getPrefix());
+        Matcher nameMatcher = namePattern.matcher(trimmedArgs);
+        Matcher moduleMatcher = modulePattern.matcher(trimmedArgs);
+        boolean hasName = nameMatcher.find();
+        boolean hasModule = moduleMatcher.find();
+
+        // Represents simply which branch it matches
+        int mode;
```
Comment (weichang18) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:60]: maybe for the mode value use enum? Eg. 0 is NAME_AND_MODULE, 1 IS NAME_ONLY, 2 IS MODULE_ONLY

**Review by weichang18 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java (hunk 1 above)
Comment (yeo-yiheng) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java:60]: Great suggestion, I'll add that in.

**Review by weichang18 — APPROVED** [review]
LGTM
